spec.html

```diff
@@ -20785,6 +21650,25 @@ <h1>Runtime Semantics: Evaluation</h1>
         1. Set _value_.[[SourceText]] to the source text matched by |ClassExpression|.
         1. Return _value_.
       </emu-alg>
+      <emu-grammar>ClassElementName : PropertyName</emu-grammar>
+      <emu-alg>
+        1. Return the result of evaluating |PropertyName|.
+      </emu-alg>
+      <emu-grammar>ClassElementName : PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. Let _privateIdentifier_ be StringValue of |PrivateIdentifier|.
+        1. Let _privateEnvRec_ be the running execution context's PrivateEnvironment.
+        1. Assert: _privateEnvRec_ has a binding for _privateIdentifier_.
+        1. If the binding for _privateIdentifier_ in _privateEnvRec_ is an uninitialized binding, then
+          1. Let _privateName_ be NewPrivateName(_privateIdentifier_).
+          1. Perform ! _privateEnvRec_.InitializeBinding(_privateIdentifier_, _privateName_).
```

Comment:
Why initialize these here, instead of during PrivateBoundIdentifiers?

Reply:
Because there's a new private name per class evaluation, so you can't do it in Static Semantics: https://rfrn.org/~shu/2018/05/02/the-semantics-of-all-js-class-elements.html#ex17

Reply:
Ah, of course.

Reply:
Though, there could be a separate runtime version which initializes them. Seems like that might be clearer, given that there's [already](https://github.com/tc39/ecma262/pull/1668#discussion_r613577918) some type confusion here.

Reply:
Eh, on further consideration, I'm not sure the thing I had in mind would be an improvement.